packages/astro/src/core/dev/util.ts

```diff
@@ -11,10 +13,27 @@ export function emoji(char: string, fallback: string) {
 	return process.platform !== 'win32' ? char : fallback;
 }
 
-export function getLocalAddress(serverAddress: string, configHostname: string): string {
-	if (configHostname === 'localhost' || serverAddress === '127.0.0.1' || serverAddress === '0.0.0.0') {
+// TODO: remove once --hostname is baselined
+export function getResolvedHostForVite(config: AstroConfig) {
+	if (config.devOptions.host === false && config.devOptions.hostname !== 'localhost') {
+		return config.devOptions.hostname;
+	} else {
+		return config.devOptions.host;
+	}
+}
+
+export function getLocalAddress(serverAddress: string, config: AstroConfig): string {
+	// TODO: remove once --hostname is baselined
+	const host = getResolvedHostForVite(config);
+	if (typeof host === 'boolean' || host === 'localhost' || serverAddress === '127.0.0.1' || serverAddress === '0.0.0.0') {
 		return 'localhost';
 	} else {
 		return serverAddress;
 	}
 }
+
+export function shouldNetworkBeExposed(config: AstroConfig) {
+	// TODO: remove once --hostname is baselined
+	const host = getResolvedHostForVite(config);
```

Comment:
does this handle if `host` is `"localhost"` or `"127.0.0.1"`?

Reply:
It actually exposes on the network whenever `host` has a value. This mirrors Vite's `--host` flag. However, I agree that's kind of confusing! Happy to avoid exposing for these 2 values.

Reply:
Updated! [See here](https://github.com/withastro/astro/pull/2760/commits/f206ec5a400444970c6006b836077dd88e61c177)